packages/webpack-extraction-plugin/src/GriffelCSSExtractionPlugin.ts

```diff
@@ -134,58 +142,85 @@ export class GriffelCSSExtractionPlugin {
+          const resourcePath = module.resource;
+
+          (loaderContext as SupplementedLoaderContext)[GriffelCssLoaderContextKey] = {
+            registerExtractedCss(css: string) {
+              cssByModuleMap.set(resourcePath, css);
+            },
+            getExtractedCss() {
+              const css = cssByModuleMap.get(resourcePath) ?? '';
+              cssByModuleMap.delete(resourcePath);
+
+              return css;
+            },
+          };
+        });
+      }
+
       // WHAT?
-      //   Adds a callback to the loader context
+      //   Performs module movements between chunks if SplitChunksPlugin is not enabled.
       // WHY?
-      //   Allows us to register the CSS extracted from Griffel calls to then process in a CSS module
-      const cssByModuleMap = new Map<string, string>();
-
-      NormalModule.getCompilationHooks(compilation).loader.tap(PLUGIN_NAME, (loaderContext, module) => {
-        const resourcePath = module.resource;
-
-        (loaderContext as SupplementedLoaderContext)[GriffelCssLoaderContextKey] = {
-          registerExtractedCss(css: string) {
-            cssByModuleMap.set(resourcePath, css);
-          },
-          getExtractedCss() {
-            const css = cssByModuleMap.get(resourcePath) ?? '';
-            cssByModuleMap.delete(resourcePath);
-
-            return css;
-          },
-        };
-      });
+      //   The same reason as for SplitChunksPlugin config.
+      if (!compiler.options.optimization.splitChunks) {
+        // @ Rspack compat
+        // Rspack does not support adding chunks in the same as Webpack, we force usage of "optimization.splitChunks"
+        if (IS_RSPACK) {
+          throw new Error(
+            [
+              'You are using Rspack, but don\'t have "optimization.splitChunks" enabled.',
+              '"optimization.splitChunks" should be enabled for "@griffel/webpack-extraction-plugin" to function properly.',
+            ].join(' '),
+          );
+        }
 
-      compilation.hooks.optimizeChunks.tap({ name: PLUGIN_NAME, stage: OPTIMIZE_CHUNKS_STAGE_ADVANCED }, () => {
-        // WHAT?
-        //   Performs module movements between chunks if SplitChunksPlugin is not enabled.
-        // WHY?
-        //   The same reason as for SplitChunksPlugin config.
-        if (!compiler.options.optimization.splitChunks) {
+        compilation.hooks.optimizeChunks.tap({ name: PLUGIN_NAME, stage: OPTIMIZE_CHUNKS_STAGE_ADVANCED }, () => {
           moveCSSModulesToGriffelChunk(compilation);
+        });
+      }
+
+      // WHAT?
+      //  Disconnects Griffel chunk from other chunks, so Griffel chunk cannot be loaded async. Also connects with
+      //  the main entrypoint in config.
+      // WHY?
+      //  This is scenario required by one of MS teams. Will be removed in the future.
+      if (this.attachToMainEntryPoint) {
+        // @ Rspack compat
+        // We don't support this scenario for Rspack yet.
```

Comment:
wouldn't this mean that with rspack the extracted css can be loaded async and cause style flashing?

Reply:
Nope, that trick is a specific for Teams and allows to load async instead as it breaks connections with other chunks.